android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java

```diff
@@ -54,15 +57,16 @@ public void popToRootWindow(@Kroll.argument(optional = true) Object arg)
 
 	@Override
 	@Kroll.method
-	public void close(@Kroll.argument(optional = true) Object arg)
+	public KrollPromise<Void> close(@Kroll.argument(optional = true) Object arg)
 	{
+		// FIXME: don't do this stuff sync first!
 		if (opened) {
 			opened = false;
 			popToRootWindow(arg);
 			closeWindow(windows.get(0), arg); // close the root window
-			fireEvent(TiC.EVENT_CLOSE, null);
+			fireEvent(TiC.EVENT_CLOSE, null); // FIXME: what about close promise?
```

Comment:
This might be a good oppertunity to move the `fireEvent(TiC.EVENT_CLOSE, ...` to [TiWindowProxy.java](https://github.com/appcelerator/titanium_mobile/blob/master/android/titanium/src/java/org/appcelerator/titanium/proxy/TiWindowProxy.java#L165)

Reply:
I forgot I left this loose thread here. The tests prove it's not working right either. The issue here is we synchronously close the window and then call up to `super`, which sees that `opened == false` and returns a rejected `Promise`.

We really want to "skip" up the hierarchy to call `TiWindowProxy.open()` (not `WindowProxy.open()`). Maybe I can use `handleOpen()` overriding to fix this?